ext/snmp/snmp.c

```diff
@@ -1827,48 +1830,46 @@ PHP_SNMP_LONG_PROPERTY_READER_FUNCTION(exceptions_enabled)
 /* {{{ */
 static int php_snmp_write_info(php_snmp_object *snmp_object, zval *newval)
 {
-	php_error_docref(NULL, E_WARNING, "info property is read-only");
+	zend_throw_error(NULL, "info property is read-only");
 	return FAILURE;
 }
 /* }}} */
 
 /* {{{ */
 static int php_snmp_write_max_oids(php_snmp_object *snmp_object, zval *newval)
 {
-	int ret = SUCCESS;
 	zend_long lval;
 
 	if (Z_TYPE_P(newval) == IS_NULL) {
 		snmp_object->max_oids = 0;
-		return ret;
+		return SUCCESS;
 	}
 
 	lval = zval_get_long(newval);
 
-	if (lval > 0) {
-		snmp_object->max_oids = lval;
-	} else {
-		php_error_docref(NULL, E_WARNING, "max_oids should be positive integer or NULL, got " ZEND_LONG_FMT, lval);
+	if (lval <= 0) {
+		zend_value_error("max_oids must be greater than 0 or NULL");
```

Comment:
Is there an existing function instead of `zval_get_long()` which behaves more consistently with usual ZPP rules? E.g. won't return the number of array items when the `newval` is an array. :D

otherwise, I would say something like `Property SNMP::$max_oids must be greater than 0`. AFAIR, `null` is neither mentioned explicitly in case of nullable parameters in similar situations (e.g. `shm_attach()`).

Reply:
I don't think there is, but NULL is handled explicitly above, and as this is a property hook I have no clue about a better approach for handling the type.

Reply:
This looks okay to me, but possibly lowercase `null` would be more common now?